javascript/javascript1/week3/notepad.js

```diff
@@ -0,0 +1,89 @@
+const notes = [];
+
+//save a note
+function saveNote(content, id) {
+  const index = notes.findIndex((note) => note.id === id);
+  if (index !== -1) {
+    console.log(`A note with ID ${id} already exists. Please use a unique ID`);
+  } else {
+    const note = { content, id };
+    notes.push(note);
+  }
+}
+
+saveNote('Pick up groceries', 1);
+saveNote('Do laundry', 2);
+saveNote('Do laundry', 2);
+console.log(notes); //[{ content: 'Pick up groceries', id: 1 }, { content: 'Do laundry', id: 2 }]
+
+//get a note
+function getNote(id) {
+  if (id === undefined || typeof id !== 'number') {
+    console.log('Error: The ID is not specified or the ID is not a number');
+    return;
+  }
+  for (let i = 0; i < notes.length; i++) {
+    const note = notes[i];
+    if (note.id === id) return note;
+  }
+  console.log(`Note not found for ID ${id}`);
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+const firstNote1 = getNote();
+console.log(firstNote1); // Error: The ID is not specified or the ID is not a number
+const firstNote2 = getNote('2');
+console.log(firstNote2); // Error: The ID is not specified or the ID is not a number
+const firstNote3 = getNote(3);
+console.log(firstNote3); // Note not found for ID 3
+
+//Log out notes
+function logOutNotesFormatted() {
+  let output = '';
+  for (const note of notes) {
+    output += `The note with id: ${note.id}, has the following note text: ${note.content}`;
+    if (notes.indexOf(note) !== notes.length - 1) output += '\n';
```

Comment:
It took me a moment to understand this code. It seems like a lot of logic for removing an extra empty line, but don't think there is an easier way. I would have left it with the extra.
Consider adding comments to explain some of the logic. It can save you time when you come back to the files in a few months, or they might help others.